Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/CertficateBean.java

```diff
@@ -266,6 +272,58 @@ private void importSelectedCertificate() {
         }
     }
 
+    public void updateSelectedCertificate() {
+        LOG.info("updateSelectedCertificate");
+        if (selectedTSCertificate != null){
+            String alias = selectedTSCertificate.getAlias();
+            if (StringUtils.isNotBlank(alias) && !service.isAliasInUse(alias, service.fetchTrustStores())) {
+                try {
+                    boolean updateStatus = service.updateCertificateTS(oldAlias, selectedTSCertificate);
+                    postUpdateAction(updateStatus, VIEW_CERT_ERR_MSG);
+                } catch (CertificateManagerException ex) {
+                    LOG.error("Unable to update certificate {}", ex.getLocalizedMessage(), ex);
+                    HelperUtil.addMessageError(VIEW_CERT_ERR_MSG, ex.getLocalizedMessage());
+                }
+            } else {
+                selectedTSCertificate.setAlias(oldAlias);
+                HelperUtil.addMessageError(VIEW_CERT_ERR_MSG, "Alias already in use");
+            }
+        }
+    }
+
+    public void updateSelectedCertificateKS() {
+        LOG.info("updateSelectedCertificate");
+        if (selectedTSCertificate != null) {
+            String alias = selectedTSCertificate.getAlias();
+            if (StringUtils.isNotBlank(alias) && !service.isAliasInUse(alias, service.fetchKeyStores())) {
+                try {
+                    boolean updateStatus = service.updateCertificateKS(oldAlias, selectedTSCertificate);
+                    postUpdateAction(updateStatus, VIEW_CERT_ERROR_MSG_KS);
+                } catch (CertificateManagerException ex) {
+                    LOG.error("Unable to update certificate {}", ex.getLocalizedMessage(), ex);
+                    HelperUtil.addMessageError(VIEW_CERT_ERROR_MSG_KS, ex.getLocalizedMessage());
+                }
+            } else {
+                selectedTSCertificate.setAlias(oldAlias);
+                HelperUtil.addMessageError(VIEW_CERT_ERROR_MSG_KS, "Alias already in use");
+            }
+        }
+    }
+
+    /**
+     * @param updateStatus
+     */
+    private void postUpdateAction(boolean updateStatus, String uiElement) {
+        if (updateStatus) {
+            truststores = service.refreshTrustStores();
```

Comment:
both truststore and keystore are getting refreshed here. You only need to refresh one.